StandardSocketsHttpHandler/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -363,8 +364,8 @@ public async Task<HttpResponseMessage> SendAsyncCore(HttpRequestMessage request,
             HttpMethod normalizedMethod = HttpMethodUtils.Normalize(request.Method);
             bool hasExpectContinueHeader = request.HasHeaders() && request.Headers.ExpectContinue == true;
 
-            Debug.Assert(!_canRetry);
             _canRetry = true;

```

Comment:
In the original PR _canRetry was set here to false, In your PR it remains true, is that on purpose?
After your PR _canRetry will always be true

Reply:
Thanks for noticing this, yes, this looks like a bug. I'll take a look at it later.